src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addAnimal(animal) {
+    Animal.alive.push(animal);
+  }
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.addAnimal(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore instanceof Herbivore) {
```

Comment:
You should check on instances of Herbivore at first and then on hidden

Reply:
Does it affect performance? && searches for first falsy value anyway. So since only herbivores have hidden ability it doens't really matter. Thought it would matter if the hidden ability was added to carnivores too.